### PR TITLE
CLI tool command migrate-database executing in dry-run mode insert entries into table FLY_HFJ_MIGRATION

### PR DESCRIPTION
**Pull request by epeartree**
**Background:**
when testing database migration with the cli tool migrate-database command executing in dry-run mode, entries are added to table FLY_HFJ_MIGRATION.

**What was done:**
- added test;
- preventing inserts to the migration table when the executing task is operating in dry-run mode;
- added changelog;

### CODE REVIEW

**Review by lukedegruchy — APPROVED** [review]

File: hapi-fhir-cli/hapi-fhir-cli-api/src/test/java/ca/uhn/fhir/cli/HapiFlywayMigrateDatabaseCommandTest.java [hunk 1]
```diff
@@ -160,6 +163,7 @@ public void testMigrateFrom340_dryRun() throws IOException, SQLException {
 		// Verify that foreign key FK_SEARCHRES_RES on HFJ_SEARCH_RESULT exists
 		foreignKeys = JdbcUtils.getForeignKeys(connectionProperties, "HFJ_RESOURCE", "HFJ_SEARCH_RESULT");
 		assertTrue(foreignKeys.contains("FK_SEARCHRES_RES"));
+		assertFalse(JdbcUtils.getTableNames(connectionProperties).contains(SchemaMigrator.HAPI_FHIR_MIGRATION_TABLENAME));
```
Comment (michaelabuckley) [inline hapi-fhir-cli/hapi-fhir-cli-api/src/test/java/ca/uhn/fhir/cli/HapiFlywayMigrateDatabaseCommandTest.java:166]: This test looks ok.  But there's another scenario: existing database (so the migration table exists) -- verify no new rows.  Would that test be hard to create?

File: hapi-fhir-sql-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/HapiMigrator.java [hunk 2]
```diff
@@ -188,7 +188,9 @@ private void preExecute(BaseTask theTask) {
 	}
 
 	private void postExecute(BaseTask theNext, StopWatch theStopWatch, boolean theSuccess) {
-		myHapiMigrationStorageSvc.saveTask(theNext, Math.toIntExact(theStopWatch.getMillis()), theSuccess);
+		if (!theNext.isDryRun()) {
```
Comment (michaelabuckley) [inline hapi-fhir-sql-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/HapiMigrator.java:191]: I wonder if line 222 (createMigrationTableIfRequired();) should have the same guard.

**Review by michaelabuckley — APPROVED** [review]

File: hapi-fhir-cli/hapi-fhir-cli-api/src/test/java/ca/uhn/fhir/cli/HapiFlywayMigrateDatabaseCommandTest.java (hunk 1 above)
Comment (epeartree) [reply hapi-fhir-cli/hapi-fhir-cli-api/src/test/java/ca/uhn/fhir/cli/HapiFlywayMigrateDatabaseCommandTest.java]: ur right,  i modified the test to simulate a common scenario, that is
1- some migration has is already create/updated a schema and the migration table has entries;
2- a user want's to test a migration;

so i've added seeding of the migration table in the test pre-condition.

File: hapi-fhir-sql-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/HapiMigrator.java (hunk 2 above)
Comment (epeartree) [reply hapi-fhir-sql-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/HapiMigrator.java:191]: i agree.  made the fix and added test.

**Pull request merged**